samples/bluetooth/mesh_provisioner/src/main.c

```diff
@@ -138,34 +140,67 @@ static void configure_node(struct bt_mesh_cdb_node *node)
 	/* Add Application Key */
 	err = bt_mesh_cfg_app_key_add(net_idx, node->addr, net_idx, app_idx,
 				      key->keys[0].app_key, NULL);
-	if (err < 0) {
+	if (err) {
 		printk("Failed to add app-key (err %d)\n", err);
 		return;
 	}
 
-	/* Bind to Health model */
-	err = bt_mesh_cfg_mod_app_bind(net_idx, node->addr, node->addr, app_idx,
-				       BT_MESH_MODEL_ID_HEALTH_SRV, NULL);
-	if (err < 0) {
-		printk("Failed to bind app-key (err %d)\n", err);
+	/* Get the node's composition data and bind all models to the appkey */
+	err = bt_mesh_cfg_comp_data_get(net_idx, node->addr, 0, &status, &buf);
+	if (err || status)  {
+		printk("Failed to get Composition data (err %d, status: %d)\n",
+		       err, status);
 		return;
 	}
 
-	pub.addr = 1;
-	pub.app_idx = key->app_idx;
-	pub.cred_flag = false;
-	pub.ttl = 7;
-	pub.period = BT_MESH_PUB_PERIOD_10SEC(1);
-	pub.transmit = 0;
-
-	err = bt_mesh_cfg_mod_pub_set(net_idx, node->addr, node->addr,
-				      BT_MESH_MODEL_ID_HEALTH_SRV, &pub,
-				      &status);
-	if (err < 0) {
-		printk("mod_pub_set %d, %d\n", err, status);
+	err = bt_mesh_comp_page_0_get(&buf, &comp);
+	if (err) {
+		printk("Unable to parse composition data (err: %d)\n", err);
 		return;
 	}
 
+	elem_addr = node->addr;
+	while (!bt_mesh_comp_elem_pull(&comp, &elem)) {
```

Comment:
`== 0` would be clearer to not make the reader think the function returns bool.

Reply:
Alternatively, rethink the return value type (bool, enum, give back the elem pointer, etc)

Reply:
Now returning a pointer, to avoid returning bool to indicate errors.